src/codegen/llvm_index.rs

```diff
@@ -11,17 +11,31 @@ use std::collections::HashMap;
 pub struct LlvmTypedIndex<'ink> {
     parent_index: Option<&'ink LlvmTypedIndex<'ink>>,
     type_associations: HashMap<String, BasicTypeEnum<'ink>>,
+    pou_type_associations: HashMap<String, BasicTypeEnum<'ink>>,
     initial_value_associations: HashMap<String, BasicValueEnum<'ink>>,
     loaded_variable_associations: HashMap<String, PointerValue<'ink>>,
     implementations: HashMap<String, FunctionValue<'ink>>,
     constants: HashMap<String, BasicValueEnum<'ink>>,
 }
 
 impl<'ink> LlvmTypedIndex<'ink> {
+    pub fn new() -> LlvmTypedIndex<'ink> {
```

Comment:
you can add `#[derive(Default)]` on `LlvmTypedIndex` and then just call `LlvmTypedIndex::default()` or even `Default::default()` when you need a new instance